lib/widgets/migrate_legacy_tokens_dialog.dart

```diff
@@ -0,0 +1,191 @@
+}
+
+class _MigrateLegacyTokensDialogState extends State<MigrateLegacyTokensDialog> {
+  Widget _content = Row(
+    mainAxisAlignment: MainAxisAlignment.center,
+    children: <Widget>[CircularProgressIndicator()],
+  );
+
+  TapGestureRecognizer _tabRecognizer;
+  var problem; // I.e. Error or Exception
+
+  @override
+  void initState() {
+    super.initState();
+    _tabRecognizer = TapGestureRecognizer()
+      ..onTap = () => _launchUri(widget._githubLink);
+    _migrateTokens();
+  }
+
+  @override
+  void dispose() {
+    _tabRecognizer?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BackdropFilter(
+      filter: ImageFilter.blur(sigmaX: 5, sigmaY: 5),
+      child: AlertDialog(
+        title: Text('Migrating tokens from previous version'),
+        content: _content,
+        actions: <Widget>[
+          Visibility(
+            visible: problem != null,
+            child: RaisedButton(
+              child: Text('Save'),
+              onPressed: () async {
+                final params = SaveFileDialogParams(
+                  sourceFilePath: null,
+                  fileName: 'privacyIDEA_authenticator_${DateTime.now()}.log',
+                  data: utf8.encode('$problem'),
+                );
+
+                await FlutterFileDialog.saveFile(params: params);
+              },
+            ),
+          ),
+          Visibility(
+            visible: problem != null,
+            child: RaisedButton(
+              onPressed: () =>
+                  Clipboard.setData(new ClipboardData(text: '$problem')),
+              child: Text('Copy'),
+            ),
+          ),
+          RaisedButton(
+            child: Text(Localization.of(context).dismiss),
+            onPressed: () => Navigator.pop(context),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _launchUri(Uri link) async {
+    String uri = link.toString();
+    if (await canLaunch(uri)) launch(uri);
+  }
+
+  void _migrateTokens() async {
+    List<Widget> children = [];
+
+    // Load legacy tokens and add them to the storage.
+    log('Attempt to load legacy tokens.',
+        name: 'migrate_legacy_tokens_dialog.dart');
+    try {
+      List<Token> existingTokens = await StorageUtil.loadAllTokens();
+      List<Token> legacyTokens = await StorageUtil.loadAllTokensLegacy();
+
+      for (Token t in legacyTokens) {
+        // TODO? Prevent migrating token if it already was migrated!
+        // This would likely overwrite the existing token.
+//        if ((t is OTPToken &&
+//                existingTokens
+//                    .whereType<OTPToken>()
+//                    .any((e) => e.secret == t.secret && e.type == t.type)) ||
+//            (t is PushToken &&
+//                existingTokens
+//                    .whereType<PushToken>()
+//                    .any((e) => e.serial == t.serial))) {
+//          continue;
+//        }
+        await StorageUtil.saveOrReplaceToken(t);
```

Comment:
For testing purposes you can add 
```suggestion
        await StorageUtil.saveOrReplaceToken(t);
        throw Exception('Something went wrong!');
```